arwen/host/execution.go

```diff
@@ -532,3 +568,336 @@ func (host *vmHost) createETHCallInput() []byte {
 
 	return newInput
 }
+
+/**
+ * processAsyncInfo takes a list of async calls and for each of them, if the code exists and can be processed on this
+ *  host it will. For all others, a vm output account is generated for an actual async call.
+ *  Given the fact that the generated async calls that remain pending will be saved on storage, the processing is
+ *  done in two steps in order to correctly use all remaining gas. We first split the gas as specified by the developer,
+ *  then we save the storage, then we split again the gas to calls that leave this shard.
+ *
+ * returns a list of pending calls (the ones that should be processed on other hosts)
+ */
+func (host *vmHost) processAsyncInfo(asyncInfo *vmcommon.AsyncContextInfo) (*vmcommon.AsyncContextInfo, error) {
```

Comment:
The async-related methods of `vmHost` should be moved to `asyncCall.go`. All functions in `asyncCall.go` are in fact methods of `vmHost`.

Reply:
+1

Reply:
Done